### PR TITLE
Add code_search and dependency_snapshots for RateLimits

### PR DESCRIPTION
**Pull request by rufusnufus**
resolves https://github.com/google/go-github/issues/3018

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @rufusnufus !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging after tests pass.

**Pull request merged**
